src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -372,14 +379,17 @@ private PSTaskBase()
         /// <param name="scriptBlock">Script block to run.</param>
         /// <param name="usingValuesMap">Using variable values passed to script block.</param>
         /// <param name="dollarUnderbar">Dollar underbar variable value.</param>
+        /// <param name="currentLocation">Current working directory.</param>
         protected PSTaskBase(
             ScriptBlock scriptBlock,
             Dictionary<string, object> usingValuesMap,
-            object dollarUnderbar) : this()
+            object dollarUnderbar,
+            PathInfo currentLocation) : this()
```

Comment:
Even this constructor is public in the sense that libraries linking against PowerShell can inherit from and depend on it

Reply:
It is not released yet.

Reply:
This is an experimental feature, subject to change.